services/blockstorage/block_sync.go

```diff
@@ -68,9 +68,9 @@ func (b *BlockSync) mainLoop(ctx context.Context) {
 
 	event = startSyncEvent{}
 
-	startSyncTimer := synchronization.TempUntilJonathanTimer(b.config.BlockSyncInterval(), func() {
+	startSyncTimer := synchronization.NewTrigger(b.config.BlockSyncInterval(), func() {
 		b.events <- startSyncEvent{}
-	})
+	}, false)
```

Comment:
Please create two separate constructors, one for periodical trigger and one for a one time trigger.